subsys/caf/modules/leds.c

```diff
@@ -32,29 +32,45 @@ struct led {
 	struct k_delayed_work work;
 };
 
-static struct led leds[CONFIG_CAF_LEDS_COUNT];
+#define DT_DRV_COMPAT pwm_leds
 
+#define _LED_COLOR_ID(id) 0,
 
-static void pwm_out(struct led *led, struct led_color *color)
-{
-	for (size_t i = 0; i < ARRAY_SIZE(color->c); i++) {
-		int err = pwm_pin_set_usec(led->pwm_dev,
-					   led_pins[LED_ID(led)][i],
-					   CONFIG_CAF_LEDS_BRIGHTNESS_MAX,
-					   color->c[i],
-					   0);
+#define _LED_COLOR_COUNT(id)					\
+	static const uint8_t led_colors_##id[] = {		\
+		DT_INST_FOREACH_CHILD(id, _LED_COLOR_ID)	\
+	};
+
+DT_INST_FOREACH_STATUS_OKAY(_LED_COLOR_COUNT)
+
+#define _LED_INSTANCE_DEF(id)						\
+	{								\
+		.label = DT_INST_PROP_OR(id, label, "LED_PWM_"#id),	\
```

Comment:
Do we need to provide a default value here? If not then we could use `DT_INST_PROP` macro here.

Reply:
yes because person defining pwmleds nodes in DKs dts files forgot to add a label; since there is no label you need to use default one